File: app/src/main/java/org/zimmob/zimlx/icon/BaseIconProvider.java
package org.zimmob.zimlx.icon;

import android.graphics.drawable.Drawable;
import android.widget.TextView;

public class BaseIconProvider implements IconProvider{
    @Override
    public void loadIcon(IconTargetType type, int forceSize, Object target, Object... args) {

    }

    @Override
    public void cancelLoad(IconTargetType type, Object target) {

    }

    @Override
    public boolean isGroupIconDrawable() {
        return false;
    }

    @Override
    public Drawable getDrawableSynchronously(int forceSize) {
        return null;
    }
}

File: core/src/main/java/org/zimmob/zimlx/core/util/BaseIconProvider.kt
package org.zimmob.zimlx.core.util

import android.widget.TextView

import org.zimmob.zimlx.core.interfaces.IconDrawer
import org.zimmob.zimlx.core.interfaces.IconProvider

abstract class BaseIconProvider : IconProvider {

    fun loadIconIntoIconDrawer(iconDrawer: IconDrawer, forceSize: Int, index: Int) {
        loadIcon(IconProvider.IconTargetType.IconDrawer, forceSize, iconDrawer, index)
    }

    fun cancelLoad(iconDrawer: IconDrawer) {
        cancelLoad(IconProvider.IconTargetType.IconDrawer, iconDrawer)
    }

    fun loadIconIntoTextView(tv: TextView, forceSize: Int, gravity: Int) {
        loadIcon(IconProvider.IconTargetType.TextView, forceSize, tv, gravity)
    }

    fun cancelLoad(tv: TextView) {
        cancelLoad(IconProvider.IconTargetType.TextView, tv)
    }
}
